cloud-db/acl-overview/ios.md

```diff
@@ -0,0 +1,320 @@
+        if error != nil {
+            print("Error: \(error?.localizedDescription)")
+            return
+        }
+
+        print("Success!")
+    }
+}
+```
+
+You can also set the default roles so that newly signed up user would start with these roles.
+
+```obj-c
+[auth setUserDefaultRole:@[visitor]
+              completion:^(NSError *error) {
+                if (!error) {
+                  NSLog(@"Success!");
+                } else {
+                  NSLog(@"Error: %@", error.localizedDescription);
+                }
+              }];
+```
+
+```swift
+if let visitor = visitor as SKYRole! {
+    SKYContainer.default().auth.setUserDefaultRole([visitor]) { (error) in
+        if error != nil {
+            print("Error: \(error?.localizedDescription)")
+            return
+        }
+
+        print("Success!")
+    }
+}
+```
+
+#### Assign and Revoke Roles
+
+For admin user to change roles of other users:
+
+```obj-c
+// You must login as an admin user
+NSArray<NSString *> *userIDs = @[ @"John", @"Ken" ];
+NSArray<NSString *> *rolesToRevoke = @[ @"visitor" ];
+NSArray<NSString *> *rolesToAssign = @[ @"manager" ];
+SKYAuthContainer *auth = [[SKYContainer defaultContainer] auth];
+[auth revokeRolesWithNames:rolesToRevoke
+          fromUsersWihtIDs:userIDs
+                completion:^(NSError * _Nullable revokeError) {
+                    if (revokeError) {
+                        NSLog(@"Error: %@", revokeError.localizedDescription);
+                        return;
+                    }
+
+                    NSLog(@"Roles revoked");
+                    [auth assignRolesWithNames:rolesToAssign
+                                toUsersWithIDs:userIDs
+                                    completion:^(NSError * _Nullable assignError) {
+                                        if (assignError) {
+                                            NSLog(@"Error: %@", assignError.localizedDescription);
+                                            return;
+                                        }
+
+                                         NSLog(@"Roles assigned");
+                                    }];
+                }];
+```
+
+```swift
+// You must login as an admin user
+let userIDs: [String] = ["John", "Ken"]
+let rolesToRevoke: [String] = ["visitor"]
+let rolesToAssign: [String] = ["manager"]
+let auth: SKYAuthContainer = SKYContainer.default().auth
+auth.revokeRoles(withNames: rolesToRevoke, fromUsersWihtIDs: userIDs) { (revokeError) in
+    if revokeError != nil {
+        print("Error: \(revokeError?.localizedDescription)")
+        return;
+    }
+
+    print("Roles revoked")
+    auth.assignRoles(withNames: rolesToAssign, toUsersWithIDs: userIDs) { (assignError) in
+        if assignError != nil {
+            print("Error: \(assignError?.localizedDescription)")
+            return;
+        }
+
+        print("Roles assigned")
+    }
+}
+```
+
+## Changing Default ACL Settings
```

Comment:
I think this default is very confusing with the default at `recrod-acl`.

@Steven-Chan do you think we should describe it is a SDK default VS server default?

I think this SDK default should not be used in normal new APP actually.

Reply:
> I think this SDK default should not be used in normal new APP actually.

If this is the case, I recommend either

- put this later than default at `record-acl` and maybe put this inside a `:::tips` or `:::advanced` block, or
- simply remove this thing totally

Reply:
If we use :::advance, we should provide an example usage.

For example,

Server default to all manager role can read/write a feedback of an employee. When CEO login into the system SDK default will override the server default ACL to make manager can only ready the appraisal, not modify.